contracts/voting-verifier/src/events.rs

```diff
@@ -29,12 +30,32 @@ pub struct PollStarted {
     pub source_gateway_address: String,
     pub confirmation_height: u64,
     pub expires_at: u64,
-    pub messages: Vec<EvmMessage>,
+    pub data: PollData,
     pub participants: Vec<Addr>,
 }
 
+pub enum PollData {
+    Messages(Vec<EvmMessage>),
+    WorkerSet(WorkerSetConfirmation),
+}
+
+fn get_data_attribute(data: PollData) -> (String, String) {
+    match data {
+        PollData::Messages(msgs) => (
+            "messages".into(),
+            serde_json::to_string(&msgs).expect("failed to serialize messages"),
+        ),
+        PollData::WorkerSet(worker_set_confirmation) => (
+            "worker_set_confirmation".into(),
+            serde_json::to_string(&worker_set_confirmation)
+                .expect("failed to serialize worker set confirmation"),
+        ),
+    }
+}
+
```

Comment:
suggestion: make this a `From<PollData> for Attribute` implementation and then decompose the attribute when you build the event like so:
```
let Attribute{ key, value } = other.data.into();
```